File: platforms/ios/Dezel/Dezel/Modules/View/Window.swift
/**
 * The window class.
 * @class Window
 * @since 0.1.0
 */
open class Window: View {

	//--------------------------------------------------------------------------
	// MARK: Methods
	//--------------------------------------------------------------------------

	/**
	 * Removes the window.
	 * @method remove
	 * @since 0.1.0
	 */
	open func remove() {
		self.wrapper.removeFromSuperview()
	}

	/**
	 * Finds a view from this window's hierarchy at the specified point.
	 * @method viewFromPoint
	 * @since 0.2.0
	 */
	open func viewFromPoint(x: Double, y: Double, visible: Bool = true, touchable: Bool = true) -> View? {
		return self.findViewAt(point: Point3D(x: x, y: y, z: 0.0), matrix: Transform3D(), visible: visible, touchable: touchable)
	}

	//--------------------------------------------------------------------------
	// JavaScript Functions
	//--------------------------------------------------------------------------

	/**
	 * @method jsFunction_viewFromPoint
	 * @since 0.2.0
	 * @hidden
	 */
	@objc open func jsFunction_viewFromPoint(callback: JavaScriptFunctionCallback) {

		let x = callback.argument(0).number
		let y = callback.argument(1).number
		let v = callback.argument(2).boolean
		let t = callback.argument(3).boolean

		if let view = self.viewFromPoint(x: x, y: y, visible: v, touchable: t) {
			callback.returns(view.holder)
		}
	}
}

File: platforms/android/Dezel/dezel/src/main/java/ca/logaritm/dezel/modules/view/Window.kt
package ca.logaritm.dezel.modules.view


import ca.logaritm.dezel.core.JavaScriptContext
import ca.logaritm.dezel.core.JavaScriptFunctionCallback
import ca.logaritm.dezel.extension.removeFromParent
import ca.logaritm.dezel.modules.view.geom.Point3D
import ca.logaritm.dezel.modules.view.geom.Transform3D


/**
 * The root class.
 * @class Window
 * @since 0.1.0
 */
open class Window(context: JavaScriptContext): View(context) {

	//--------------------------------------------------------------------------
	// Methods
	//--------------------------------------------------------------------------

	/**
	 * @constructor
	 * @since 0.1.0
	 */
	init {

	}

	/**
	 * Removes the root.
	 * @method remove
	 * @since 0.1.0
	 */
	public fun remove() {
		this.wrapper.removeFromParent()
	}

	/**
	 * Finds a view from this window's hierarchy at the specified point.
	 * @method viewFromPoint
	 * @since 0.2.0
	 */
	open fun viewFromPoint(x: Double, y: Double, visible: Boolean = true, touchable: Boolean = true): View? {
		return this.findViewAt(Point3D(x, y, 0.0), Transform3D(), visible, touchable)
	}

	//--------------------------------------------------------------------------
	// JavaScript Functions
	//--------------------------------------------------------------------------

	/**
	 * @method jsFunction_viewFromPoint
	 * @since 0.2.0
	 * @hidden
	 */
	@Suppress("unused")
	open fun jsFunction_viewFromPoint(callback: JavaScriptFunctionCallback) {

		val x = callback.argument(0).number
		val y = callback.argument(1).number
		val v = callback.argument(2).boolean
		val t = callback.argument(3).boolean

		val view = this.viewFromPoint(x, y, v, t)
		if (view == null) {
			return
		}

		callback.returns(view.holder)
	}
}